rollup/da_syncer/da_syncer.go

```diff
@@ -0,0 +1,53 @@
+package da_syncer
+
+import (
+	"fmt"
+
+	"github.com/scroll-tech/go-ethereum/core"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rollup/da_syncer/da"
+)
+
+var (
+	ErrBlockTooLow  = fmt.Errorf("block number is too low")
+	ErrBlockTooHigh = fmt.Errorf("block number is too high")
+)
+
+type DASyncer struct {
+	blockchain *core.BlockChain
+}
+
+func NewDASyncer(blockchain *core.BlockChain) *DASyncer {
+	return &DASyncer{
+		blockchain: blockchain,
+	}
+}
+
+// SyncOneBlock receives a PartialBlock, makes sure it's the next block in the chain, executes it and inserts it to the blockchain.
+func (s *DASyncer) SyncOneBlock(block *da.PartialBlock) error {
+	currentBlock := s.blockchain.CurrentBlock()
+
+	// we expect blocks to be consecutive. block.PartialHeader.Number == parentBlock.Number+1.
+	if block.PartialHeader.Number <= currentBlock.Number().Uint64() {
+		log.Debug("block number is too low", "block number", block.PartialHeader.Number, "parent block number", currentBlock.Number().Uint64())
+		return ErrBlockTooLow
+	} else if block.PartialHeader.Number > currentBlock.Number().Uint64()+1 {
+		log.Debug("block number is too high", "block number", block.PartialHeader.Number, "parent block number", currentBlock.Number().Uint64())
+		return ErrBlockTooHigh
+	}
+
+	parentBlock := s.blockchain.GetBlockByNumber(currentBlock.Number().Uint64())
```

Comment:
Can't we directly use `currentBlock` here?

Reply:
Yeah you're right. This is probably a remnant from backporting as in the upstream branch returns `*types.Header`. I'll address this once I add the new `da-codec`.